Shopify/cli
# [WIP] Modify checkout_ui template to use the new ui_extension type

Co-authored-by Richard Powell richardo@gmail.com

DO NOT MERGE until we have a `@shopify/ui-extensions-react@2023-01` package published

### WHY are these changes introduced?

Fixes https://github.com/Shopify/ui-extensions-private/issues/2019

### WHAT is this pull request doing?

- Filter out `ui_extension` on the generate extension prompt and remove the template for `ui_extension` as it's currently hardcoded to Checkout. In the future we will have multiple templates for `ui_extension` but it's out of scope for this PR
- Update checkout_ui_extension template to use the new ui_extension config and library
- Update fixtures/app/extensions/checkout-ui to reflect the new template

### How to test your changes?

1. Run `pnpm shopify app generate extension --path fixtures/app`
1. From the prompt, select `Checkout UI`
1. Verify that the extension is generated successfully
1. Run `pnpm shopify app dev --path fixtures/app` and open the Preview Url
1. Verify that both checkout-ui extension and your new extension are running as ui_extensions - they each should have a preview link for the extension point

<img width="1867" alt="image" src="https://user-images.githubusercontent.com/29458473/216435599-2c723b98-7f00-4975-80c0-8806c259c3f1.png">


### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
